main.py

```diff
@@ -186,62 +186,106 @@ async def set_playing(status):
     await client.change_presence(activity=discord.Game(name=status))
 
 
-async def give_random(channel, number_of_issues):
-    """
-    @TODO: improve algorithm to save queries
-    1. get first batch of github_max (100), find total number
-    2. if total < requested, return all
-    3. Generate requested random numbers
-    4. get issues till max(generated_list)
-    5. return them
-    """
-    await set_playing('Finding Issues')
+def get_params(command):
```

Comment:
While I appreciate the effort, you've split 3 functions based on the command. That's not good.

Merely modifying `request_q` and the way title was worded would have been sufficient. You could've called `request_q` with 2 different labels and it would have worked just fine

Reply:
I'll undo the changes if you want, but I wanted to know why exactly it isn't good? Is it a design issue or it'll be difficult to handle later, what problems may arise?

> Merely modifying request_q and the way title was worded would have been sufficient.

That's true, but I noticed code repetition in the random generator too. Plus, I was also thinking if we wanted to add `needs: feedback issues` (and more filters, if needed), it would be better to have a general function of some sort.

So, do I go back to the previous implementation? Maybe you suggest an even better alternative structure.

Reply:
> Is it a design issue

Yes. The details for each command is split across multiple functions. That's a bad sign in itself.

I didn't notice the repetition in random-generator before, but maybe restructure in the following way:
1. each command has it's own function which supplies the details (issue/pr, author, include/exclude label)
2. call the search result generator
3. transform the search results (aka replace them with details via another API call)
4. perform some operation on the search result (aka: random select, top select, etc.)

2-4 is a pipeline which can be written as a separate function, with the transformer and operations being independent functions. The higher order function are sent to the pipeline in step 1.

Don't follow this blindly, since I didn't spend a lot of time on the code. The code is very young so feel free to experiment with control inversion or generators or co-routines, etc.

Reply:
Let's get this one going soon if possible

Reply:
Sure.

Reply:
> each command has it's own function which supplies the details (issue/pr, author, include/exclude label)

For this, I was thinking of treating it as configurations (dictionaries). Can I do the configurations in a separate file? If yes, then will a `.py` be fine? Or `.json`, `.yaml`, something like that?

Reply:
Converting "Sorting by author" an option in the same command (or derivating using prefix/suffix, additional term, etc.) would be nice too

Reply:
Do you mean for the `needs: feedback` label, sort by the PR authors' names as an additional option, on top of the `order by oldest`?

Reply:
I meant something similar to the difference between the query and review commands

Reply:
But a PR doesn't need the feedback of an individual, right? If yes, then I guess those will be the reviewers.

Reply:
Oh yeah. Scrap that thought. My bad